render/fx_renderer/fx_pass.c

```diff
@@ -945,14 +1027,13 @@ void fx_render_pass_add_blur(struct fx_gles_render_pass *pass,
 	struct fx_texture *blur_texture = fx_get_texture(wlr_texture);
 	blur_texture->has_alpha = true;
 
+	bool should_ignore_transparent = fx_options->ignore_transparent &&
+			tex_options->base.texture;
+
 	// Get a stencil of the window ignoring transparent regions
-	if (fx_options->ignore_transparent && fx_options->tex_options.base.texture) {
+	if (should_ignore_transparent) {
 		stencil_mask_init();
-
-		struct fx_render_texture_options tex_options = fx_options->tex_options;
-		tex_options.discard_transparent = true;
-		fx_render_pass_add_texture(pass, &tex_options);
-
+		fx_render_pass_add_discard_transparent(pass, &tex_options->base);
```

Comment:
Wouldn't it be better to just create a `fx_render_pass_add_blur_texture` or something which does what these lines + lines 959->969? Then we should be able to remove the stenciling right?